commands/cluster/cluster.go

```diff
@@ -17,35 +17,113 @@ package commands
 
 import (
 	"fmt"
+	"strings"
 
 	"github.com/spf13/cobra"
 
+	"github.com/hazelcast/hazelcast-commandline-client/config"
+	hzcerror "github.com/hazelcast/hazelcast-commandline-client/errors"
 	"github.com/hazelcast/hazelcast-commandline-client/internal"
 )
 
-const invocationOnCloudErrorMessage = "Cluster operations on cloud are not supported. Checkout https://github.com/hazelcast/hazelcast-cloud-cli for cluster management on cloud."
-
-var ClusterCmd = &cobra.Command{
-	Use:   "cluster {get-state | change-state | shutdown | query} [--state new-state]",
-	Short: "administrative cluster operations",
-	Long:  `administrative cluster operations which controls a Hazelcast cluster by manipulating its state and other features`,
-	RunE: func(cmd *cobra.Command, args []string) error {
-		hzConf, err := internal.MakeConfig()
-		if err != nil {
-			return err
-		}
-		// check if it is cloud invocation
-		if hzConf.Cluster.Cloud.Token != "" {
-			fmt.Println(invocationOnCloudErrorMessage)
+const invocationOnCloudInfoMessage = "Cluster operations on cloud are not supported. Checkout https://github.com/hazelcast/hazelcast-cloud-cli for cluster management on cloud."
+
+func New() *cobra.Command {
+	cmd := cobra.Command{
+		Use:   "cluster {get-state | change-state | shutdown | query} [--state new-state]",
+		Short: "administrative cluster operations",
+		Long:  `administrative cluster operations which controls a Hazelcast cluster by manipulating its state and other features`,
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			conf := config.FromContext(cmd.Context())
+			// check if it is a cloud invocation
+			if conf.Cluster.Cloud.Token != "" {
+				return hzcerror.NewLoggableError(nil, invocationOnCloudInfoMessage)
+			}
+			return nil
+		},
+	}
+	cmd.AddCommand(
+		NewGetState(),
+		NewChangeState(),
+		NewShutdown(),
+		NewVersion())
+	return &cmd
+}
+
+func NewGetState() *cobra.Command {
```

Comment:
`NewGetState`, `NewShutdown` and `NewVersion` are almost identical. We can gather them in a good block of code ?

Reply:
I cannot see what constraints that we will be encountered when we create them as dynamic. But I'm the side of dynamic creation.

Reply:
Thank you both for the feedback. Fixed at https://github.com/hazelcast/hazelcast-commandline-client/pull/47/commits/dd8bd57e2d0d1b3c924b63a24213a56b2a69dd5f as you suggested